src/zip_node.js

```diff
@@ -30,6 +45,42 @@ const zipNodeJs = async function ({ srcFiles, destPath, filename, mainFile, plug
   await endZip(archive, output)
 }
 
+// Zip a Node.js function file with esbuild
+const zipNodeJsWithEsbuild = async function ({
+  destFolder,
+  destPath,
+  externalModules = [],
+  filename,
+  mainFile,
+  pluginsModulesPath,
+}) {
+  const bundledFilePath = join(destFolder, filename)
+
+  await esbuild.build({
+    bundle: true,
+    entryPoints: [mainFile],
+    external: externalModules,
+    outfile: bundledFilePath,
+    platform: 'node',
+    target: ['es2017'],
+  })
+
+  const { archive, output } = startZip(destPath)
+  const { srcFile, stat } = await addStat(bundledFilePath)
+
+  addEntryFile(destFolder, archive, filename, bundledFilePath)
+
+  zipJsFile({ srcFile, destFolder, pluginsModulesPath, archive, stat })
+
+  await endZip(archive, output)
+
+  try {
```

Comment:
Should those lines be in a `try {} finally {}` block so that the `bundleFilePath` is deleted when an exception is thrown during zipping?

Reply:
That's a good call. I added the `try`/`catch` around `pUnlink` because I don't think we should throw if we fail to delete the temporary fail. If we put these two concepts together, we end up with:

```js
  try {
    await endZip(archive, output)
  } finally {
    try {
      await pUnlink(bundledFilePath)
    } catch (_) {
      // no-op
    }
  }
```

🤔  What do you think?

Reply:
Yes, something like this.
I am wondering whether, to be on the safe side, the other parts of the zipping logic (`addStat()`, `addEntryFile()`, `zipJsFile()`) should be wrapped in it too?

Note: the `try`/`catch` around `pUnlink()` definitely makes sense. :+1:

Reply:
Agreed! I'll update it. Thanks! 🙌🏻